main.go

```diff
@@ -148,13 +146,28 @@ func main() {
 		app.autostartEnabled = true
 	}
 
-	hnsErrCh := make(chan error)
 	serverErrCh := make(chan error)
 	onBoardingFilename := path.Join(app.config.Path, "init")
 	onBoarded := onBoardingSeen(onBoardingFilename)
 
+	hnsdPath, err := getProcPath()
+	if err != nil {
+		log.Fatal(err)
+	}
+	hnsdPath = path.Join(hnsdPath, "/hnsd")
+
+	hnsdCheckpointPath := ""
+	if hnsdPath == "" {
+		log.Fatal("path to hnsd is not provided")
+	}
+	if hnsdCheckpointPath == "" {
+		home, _ := os.UserHomeDir() //above already fails if it doesn't exist
+		hnsdCheckpointPath = path.Join(home, ".hnsd")
+	}
+
+	sync.GetRoots(hnsdPath, app.config.Path, hnsdCheckpointPath)
```

Comment:
I see the tray icon takes a while to show up after starting fingetip. Is this line blocking the tray icon? This can maybe go after whatever makes it show up so users know it's started.

Reply:
Made it the other way: tray is shown before sync, also the syncing status is shown correctly on the page.